### ISSUE TITLE
docs: Supported workflow systems

### ISSUE THREAD

**Issue by tiborsimko** [reanahub/docs.reana.io#101]
Populate pages such as:

- https://docs.reana.io/running-workflows/supported-systems/yadage/
- https://docs.reana.io/running-workflows/supported-systems/cwl/
- https://docs.reana.io/running-workflows/supported-systems/serial/
- https://docs.reana.io/running-workflows/supported-systems/snakemake/

These pages should have:

- a small description about each workflow engine, as in Snakemake.
- link to homepage where readers can learn more
- link to user guide or a tutorial

Moreover, the page should introduce new information:

- any notes how this engine is used in REANA, e.g. some features that are not supported

- any links to REANA blogs that describe yadage-specific or snakemake-specific posts of interest;

- the matrix of versions supported in each REANA version, such as:

```
REANA  |  Yadage   |   Packtivity
-------|-----------|-------------
  0.8  |   0.20.1  |   0.14.24
  0.7  |   0.20.1  |   0.13.10
  0.6  |   0.20.0  |   0.12.10
  ...
  0.1  |   0.19.5  |   0.10.01
 ```

This table will help users to give a quick overview of what they can expect from each REANA version they are dealing with.

The "historical" information such as REANA version 0.3 is perhaps not terribly necessary, but I'd gather it anyway so that we are "complete" here.  It'll be good for image and for foretelling future.

TL;DR The main goal of this "Workflow-engine-X-in-REANA" page is to serve as a single stop-shop entry page for experienced CWL or Snakemake users to see in one place all the important information on how their favourite workflow engine is supported in REANA, what Yadage version they can use, whether there are any non-implemented features should there be such thing (think CWL matrix), and, alternatively, whether REANA offers any cool novelties such as Snakemake-on-HTC-and-HPC.
